spec/integration_spec.rb

```diff
@@ -51,34 +40,73 @@ def tables
 
       it 'sends the output to the stdout' do
         expect($stdout).not_to receive(:puts)
-        ActiveRecord::Migrator.new(direction, [migration_fixtures], 1).migrate
+        ActiveRecord::Migrator.new(direction, migration_fixtures, 1).migrate
       end
     end
   end
 
   context 'when ActiveRecord is loaded' do
+    let(:db_config) { Configuration.new }
+
     it 'reconnects to the database using PerconaAdapter' do
-      ActiveRecord::Migrator.new(direction, [migration_fixtures], 1).migrate
-      expect(ActiveRecord::Base.connection_pool.spec.config[:adapter]).to eq('percona')
+      ActiveRecord::Migrator.new(direction, migration_fixtures, 1).migrate
+      expect(ActiveRecord::Base.connection_pool.spec.config[:adapter])
+        .to eq('percona')
+    end
+
+    context 'when a username is provided' do
+      before do
+        ActiveRecord::Base.establish_connection(
+          adapter: 'percona',
+          host: 'localhost',
+          username: 'root',
```

Comment:
could be, but we need to find a way to have another user in the DB. I didn't want this tests to get too "dirty".